var/spack/repos/builtin/packages/suite-sparse/package.py

```diff
@@ -65,19 +68,20 @@ def install(self, spec, prefix):
             'CC=%s' % self.compiler.cc,
             'CXX=%s' % self.compiler.cxx,
             'F77=%s' % self.compiler.f77,
-            'CUDA_ROOT     =',
-            'GPU_BLAS_PATH =',
-            'GPU_CONFIG    =',
-            'CUDA_PATH     =',
-            'CUDART_LIB    =',
-            'CUBLAS_LIB    =',
-            'CUDA_INC_PATH =',
-            'NV20          =',
-            'NV30          =',
-            'NV35          =',
-            'NVCC          = echo',
-            'NVCCFLAGS     =',
+            # They "autodetect" CUDA before CUDA_PATH is checked
+            # setting CUDA=no disables them using `which nvcc`,
+            # so that we can use our spack managed cuda path.  If
+            # +cuda we set CUDA_PATH, and the other CUDA specific
+            # libraries are setup by suite-sparse.  See the file
+            # SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk
+            'CUDA=no'
```

Comment:
Ok so this is was what I was shooting for:

1. They have some code in there that comes in and **first** does `which nvcc`.  If they find it then they hard-code `CUDA_PATH` to be the result of that, up a couple directories.
    - You cannot prevent them from doing this, unless you set `CUDA=no`
2. You also specify `CUDA_PATH` to what it should be.  Therefore even though their `which nvcc` stuff never executes, we use the CUDA we want.

Without this, say if you had multiple CUDA installed or something, it will always use the one that shows up in the path first.  Similarly, if you have CUDA installed but don't want it...you need to set `CUDA=no` explicitly otherwise it will still `which nvcc` and use it (they're `+cuda` by default)

Does that make sense?  You _always_ `CUDA=no`, and then set the thing that "not `CUDA=no`" _would have_ set if `+cuda`.